LibuvSharp/UVStream.cs

```diff
@@ -212,58 +212,59 @@ public void Write(ArraySegment<byte> data, Action<Exception> callback)
 			Ensure.Success(r);
 		}
 
-        public void Write(IList<ArraySegment<byte>> buffers, Action<Exception> callback)
-        {
-            CheckDisposed();
-
-            PendingWrites++;
-
-            int n = buffers.Count;
-            GCHandle[] datagchandles = new GCHandle[n];
-            CallbackPermaRequest cpr = new CallbackPermaRequest(RequestType.UV_WRITE);
-            cpr.Callback = (status, cpr2) => {
-                foreach (var datagchandle in datagchandles) {
-                    datagchandle.Free();
-                }
-                PendingWrites--;
-
-                Ensure.Success(status, callback);
-
-                if (PendingWrites == 0) {
-                    OnDrain();
-                }
-            };
-            int r;
-            if (UV.isUnix) {
-                UnixBufferStruct[] bufs = new UnixBufferStruct[n];
-                for (var i = 0; i < n; ++i) {
-                    ArraySegment<byte> data = buffers[i];
-                    int index = data.Offset;
-                    int count = data.Count;
-                    GCHandle datagchandle = GCHandle.Alloc(data.Array, GCHandleType.Pinned);
-                    var ptr = (IntPtr)(datagchandle.AddrOfPinnedObject().ToInt64() + index);
-                    bufs[i] = new UnixBufferStruct(ptr, count);
-                    datagchandles[i] = datagchandle;
-                }
-                r = uv_write_unix(cpr.Handle, NativeHandle, bufs, n, CallbackPermaRequest.CallbackDelegate);
-            } else {
-                WindowsBufferStruct[] bufs = new WindowsBufferStruct[n];
-                for (var i = 0; i < n; ++i) {
-                    ArraySegment<byte> data = buffers[i];
-                    int index = data.Offset;
-                    int count = data.Count;
-                    GCHandle datagchandle = GCHandle.Alloc(data.Array, GCHandleType.Pinned);
-                    var ptr = (IntPtr)(datagchandle.AddrOfPinnedObject().ToInt64() + index);
-                    bufs[i] = new WindowsBufferStruct(ptr, count);
-                    datagchandles[i] = datagchandle;
-                }
-                r = uv_write_win(cpr.Handle, NativeHandle, bufs, n, CallbackPermaRequest.CallbackDelegate);
-            }
-
-            Ensure.Success(r);
-        }
-
-        public void Shutdown(Action<Exception> callback)
+		public void Write(IList<ArraySegment<byte>> buffers, Action<Exception> callback)
```

Comment:
ICollection does not implement indexed access (`data = buffers[i];`), which could be worked around of course, but I am not sure if utilizing enumerators instead of just indexing is worth the overhead - especially because provided buffers are, usually, lists or arrays anyway.